includes/pattern.inc

```diff
@@ -160,3 +163,43 @@ function _oe_bootstrap_theme_extract_carousel_image(&$item) {
   }
   return FALSE;
 }
+
+/**
+ * Implements hook_preprocess_HOOK() for content banner.
+ */
+function oe_bootstrap_theme_preprocess_pattern_content_banner(array &$variables) {
+  oe_bootstrap_theme_get_image($variables);
+}
+
+/**
+ * Get imageValueObject from variables.
+ *
+ * @param array $variables
+ *   The render array.
+ */
+function oe_bootstrap_theme_get_image(array &$variables): void {
```

Comment:
This function should be renamed or explained better, imo.
The current name is quite generic, but I assume this is only meant for a very specific "$variables" array.
Also, I doubt that "$variables" is really a render array. Or is it?
Also, the name "get_image" suggests that we are returning a value (an image), when in fact we are writing into the $variables array.

Reply:
Or perhaps we don't need this function at all, and can do it all in the preprocess.

Reply:
I see two options:
1. Change this into a function that receives the value from `$variables['image']` as parameter, and returns the new value for `$variables['image']`. (or we can store it in a different key)
2. Make this a twig function that can be called from the template: `image|bcl_image`, instead of doing it from preprocess.

@brummbar opinion?

Reply:
I think the refactoring done around this function is very wrong, we are handling two different cases at once.
Some notes aside from that:
* let's preprend `_` to "internal" functions, it's an ugly Drupalism but quite effective in this realm
* try to pass a simple parameter (like array) and return an array, instead of passing variables by reference.